Projects/UOContent/Items/Misc/OilFlask.cs

```diff
@@ -0,0 +1,98 @@
+using ModernUO.Serialization;
+using Server.Targeting;
+
+namespace Server.Items;
+
+[SerializationGenerator(0)]
+public partial class OilFlask : Item
+{
+    [Constructible]
+    public OilFlask(int amount = 1) : base(0x1C18)
+    {
+        Stackable = true;
+        Amount = amount;
+    }
+
+    public override int LabelNumber => 1027199; // oil flask
+    public virtual int FilledMessageNumber => 1150864; // You fill the lamp with oil.
+    public virtual double FillMultiplier => 1;
+
+    public override void OnDoubleClick(Mobile from)
+    {
+        from.SendLocalizedMessage(501947); // Select lantern to refuel.
+        from.BeginTarget(2, true, TargetFlags.None, OnTargetPicked);
+    }
+
+    private void OnTargetPicked(Mobile from, object targeted)
+    {
+        switch (targeted)
+        {
+            case Lantern lantern:
+                if (!lantern.Movable || (lantern.Protected && from.AccessLevel == AccessLevel.Player))
+                {
+                    from.SendLocalizedMessage(500685); // You can't use that, it belongs to someone else.
+                    return;
+                }
+
+                var wasBurning = lantern.Burning;
+                lantern.Douse();
+                lantern.Duration = Lantern.FullDuration * FillMultiplier;
+                lantern.BurntOut = false;
+
+                if (wasBurning)
+                {
+                    lantern.Ignite();
+                }
+
+                from.SendLocalizedMessage(FilledMessageNumber);
+
+                var emptyFlask = new EmptyOilFlask();
+                if (Parent is Container container)
+                {
+                    container.AddItem(emptyFlask);
```

Comment:
Is this accurate? Most empty bottles and such are created in the user's backpack, regardless of the container. Let's check with drinking potions or something similar.
Usually we use `TryDropItem` so if the backpack is full, then we can put it on the ground.

Also to note: Empty flasks are stackable. So we may need code for dropping to the world that stacks with an existing flask stack, if it isn't full. We should look into how bouncing or other mechanics handle that and copy it.